CMakeLists.txt

```diff
@@ -2185,6 +2185,34 @@ list(APPEND llext_edk_cflags ${llext_filt_flags})
 list(APPEND llext_edk_cflags ${LLEXT_APPEND_FLAGS})
 list(APPEND llext_edk_cflags ${LLEXT_EDK_APPEND_FLAGS})
 
+# Define variables for the cmake sub-process
+foreach(exported_var
+        APPLICATION_SOURCE_DIR
+        CONFIG_LLEXT_EDK_NAME
+        CONFIG_LLEXT_EDK_USERSPACE_ONLY
+        llext_edk_cflags
+        llext_edk_file
+        PROJECT_BINARY_DIR
+        WEST_TOPDIR
+        ZEPHYR_BASE
+        CONFIG_LLEXT_EDK_TARGET_INFO # uses:
+            ARCH
+            BOARD
+            BOARD_REVISION
+            BOARD_QUALIFIERS
+            NORMALIZED_BOARD_QUALIFIERS
+            NORMALIZED_BOARD_TARGET
+            SOC_NAME
+            SOC_SERIES
+            SOC_FAMILY
```

Comment:
exactly why are all those details needed ?

Shouldn't it be sufficient to have the board target ?

Or are you starting to replicate the Zephyr build system outside of Zephyr ?
If that's the case, then perhaps integrate to sysbuild is a better choice.

Reply:
> - CROSS_COMPILE_TARGET is very useful as well, as it identifies the specific compiler used and whose CFLAGS are exported by the EDK.

and this exactly proves my worries of exporting those fields like this, because not all toolchains set this variable and we make no promises. But suddenly we could need to fix "bugs" because something which is intended for internal use is being exported.

An example of a toolchain where this variable is not defined is LLVM.

Reply:
> - I have also used ARCH, BOARD, and to a lesser extent SOC_NAME, to logically group different EDKs.

Be careful on using arch for grouping, because a board could have multiple SoCs of different archs or a multi-core SoC where the cores are of different arch. Of course this depends on which level the arch is place, but Zephyr has just gone through a rework to avoid arch on high level organization :wink: